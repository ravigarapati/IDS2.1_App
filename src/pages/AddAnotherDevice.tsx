import React, {Component} from 'react';
import {View, StyleSheet, Image} from 'react-native';
import {Button, CustomText} from '../components';
import {connect} from 'react-redux';
import {userFirstLogin} from '../store/actions/HomeOwnerActions';

class AddAnotherDevice extends Component {
  addAnotherDevice = () => {
    this.props.navigation.navigate('Add', {
      addAnother: true,
    });
  };

  cancel = () => {
    this.props.navigation.navigate('HomeTabs');
    //this.props.userFirstLogin(false);
  };

  render() {
    return (
      <View style={[styles.flex1, styles.padding10]}>
        <View style={[styles.flex1, styles.container]}>
          <Image source={require('./../assets/images/addanotherdevice.png')} />
          <CustomText
            allowFontScaling={true}
            font={'medium'}
            text={'Would you like to add\nanother Appliance?'}
            size={21}
          />
        </View>
        <View style={[styles.flex1, styles.buttonSection]}>
          <Button
            type="primary"
            text={'Yes'}
            onPress={this.addAnotherDevice}
            testID="submit"
          />
          <Button
            type="secondary"
            text={'No'}
            onPress={this.cancel}
            testID="cancel"
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  flex1: {
    backgroundColor: 'white',
    flex: 1,
  },
  padding10: {
    padding: 10,
  },
  container: {
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginHorizontal: 40,
  },
  buttonSection: {
    justifyContent: 'flex-end',
    marginBottom: 20,
  },
});

const mapDispatchToProps = {
  userFirstLogin,
};

export default connect(null, mapDispatchToProps)(AddAnotherDevice);
