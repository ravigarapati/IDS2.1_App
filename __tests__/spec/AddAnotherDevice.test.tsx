import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AddAnotherDevice from '../../src/pages/AddAnotherDevice';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import {render, cleanup, fireEvent} from 'react-native-testing-library';

const mockStore = configureMockStore();
const store = mockStore({
  homeOwner: {
    energyData: {},
    deviceList: [],
    selectedDevice: {},
    contractorMonitoringStatus: false,
    fanMode: '',
    fanOnFor: '',
    fanOffFor: '',
    fanIsScheduled: '',
    fanScheduledStart: '',
    fanScheduledEnd: '',
    selectedSchedule: '0',
    selectedScheduleName: '',
    scheduleInfo: {},
    prevSelectedDevice: {},
    deviceStatus: {},
    deviceInformation: {},
    deviceDetails: {},
    heatPumpInfo: {},
    reloadHeatPumpInfo: false,
    notificationList: [],
    lastKey: {},
    notificationsCount: 0,
    lastUpdated: null,
    isTermsConditionsAccepted: null,
    hoUserAnalytics: null,
    selectedUnitName: {},
    isThermostatSelected: true,
    isUserFirstLogin: true,
    actualWeatherOnFahrenheit: true,
    hapticVibration: true,
    _24HrsFormatSelected: false,
    location: 'Alabaster',
    updatedSelectedDevice: false,
    weatherInfo: {},
    weatherInfoLocation: {},
    locationSuggestions: [],
    locationInformation: {},
    idsSelectedDeviceAccess: '',
    utilyEnergySaving: '',
    idsSelectedDevice: '',
    idsSelectedOdu: '',
    idsSelectedDeviceType: '',
    tempLocation: '',
    newDeviceInfo: {},
    previousBcc: '',
    deviceList2: [
      /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
    ],
  },
  auth: {},
});

describe('Add Another Device screen', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddAnotherDevice />
        </Provider>,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });

  test('Component to be rendered', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddAnotherDevice />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Click on add Another button', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: page => {
        currentScreen = page;
      },
    };
    const onSubmit = jest.fn();
    const rendered = render(
      <Provider store={store}>
        <AddAnotherDevice
          navigation={navigation}
          userFirstLogin={value => {}}
        />
      </Provider>,
    );
    const buttonComponent = rendered.getByTestId('submit');

    fireEvent(buttonComponent, 'press');
    //expect(buttonComponent);
    expect(currentScreen).toBe('Add');
  });

  test('Click on cancel button', () => {
    let currentScreen = '';
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      navigate: page => {
        currentScreen = page;
      },
    };
    const onSubmit = jest.fn();
    const rendered = render(
      <Provider store={store}>
        <AddAnotherDevice
          navigation={navigation}
          userFirstLogin={value => {}}
        />
      </Provider>,
    );
    const cancelComponent = rendered.getByTestId('cancel');
    fireEvent(cancelComponent, 'press');
    expect(currentScreen).toBe('HomeTabs');
  });
});
