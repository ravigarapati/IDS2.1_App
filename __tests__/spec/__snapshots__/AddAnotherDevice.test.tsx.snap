// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Another Device screen Component to be rendered 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 1,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "space-evenly",
          "marginHorizontal": 40,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/addanotherdevice.png",
        }
      }
    />
    <Text
      accessible={true}
      numberOfLines={0}
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#000000",
          },
          Object {
            "fontSize": 21,
            "lineHeight": 31.5,
          },
          Object {
            "fontFamily": "BoschSans-Medium",
          },
          Object {},
        ]
      }
    >
      Would you like to add
another Appliance?
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 1,
        },
        Object {
          "justifyContent": "flex-end",
          "marginBottom": 20,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityElementsHidden={false}
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        importantForAccessibility="auto"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginVertical": 5,
              "paddingVertical": 15,
            },
            Object {},
            Object {
              "backgroundColor": "#004975",
            },
            Object {},
            Object {
              "borderColor": "#004975",
              "borderWidth": 1,
            },
          ]
        }
        testID="submit"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Regular",
                },
                Object {},
              ]
            }
          >
            Yes
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        accessibilityElementsHidden={false}
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        importantForAccessibility="auto"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginVertical": 5,
              "paddingVertical": 15,
            },
            Object {},
            Object {},
            Object {
              "borderColor": "#004975",
              "borderWidth": 1,
            },
          ]
        }
        testID="cancel"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
                Object {
                  "color": "#004975",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Regular",
                },
                Object {},
              ]
            }
          >
            No
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
